src/Symfony/Component/DependencyInjection/Context.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Exception\BadMethodCallException;
+use Symfony\Component\DependencyInjection\Exception\ContextElementNotFoundException;
+
+/**
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ */
+final class Context
+{
+    private $elements;
+    private $locked = false;
+
+    public function __construct(array $context = array())
```

Comment:
at this point it's not very clear what `$context` is supposed to be and it might be a bit confusing to pass a context array to a `Context` class...